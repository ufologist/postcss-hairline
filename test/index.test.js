const postcss = require('postcss');
const fs = require('fs');
const path = require('path');

const plugin = require('../src/index.js');

const readFile = (filePath, ...args) => new Promise((resolve, reject) => {
    fs.readFile(path.resolve(__dirname, filePath), ...args, (err, data) => {
        if (err) {
            reject();
        } else {
            resolve(data);
        }
    });
});

async function run(input, opts) {
    return postcss([plugin(opts)]).process(input).then((result) => {
               expect(result.css).toMatchSnapshot();
               expect(result.warnings().length).toBe(0);
           });
}

it('transform border with border-radius', async () => {
    const input = await readFile('./case/border-width-radius.css');
    run(input);
});