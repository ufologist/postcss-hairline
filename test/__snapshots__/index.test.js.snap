// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform border with border-radius 1`] = `
".px {
    background-color: #aaa;
    height: 100px; /* hairline */ /* hairline */
    border-radius: 10px;
    /* no */
    position: relative;
}

.rem {
    background-color: #aaa;
    height: 100px; /* hairline */
    border-radius: 0.5rem; /* hairline */ position: relative;
}

.before {
    background-color: #aaa;
    height: 100px; /* hairline-before */
    border-radius: 0.5rem; /* hairline */ position: relative;
}

.after {
    background-color: #aaa;
    height: 100px; /* hairline-after */ /* hairline */
    border-radius: 10px;
    /* no */
    position: relative;
}

.only-border {
    background-color: #aaa;
    height: 100px; /* hairline */ position: relative;
}

.px::after {
    border: 1px solid red;
    /* no */
    border-radius: 20px;
    /* no */
}

.rem::after {
    border: 1px solid red;
    /* no */
    border-radius: 1rem;
    /* no */
}

.before::before {
    border: 1px solid red;
    /* no */
    border-radius: 1rem;
    /* no */
}

.after::after {
    border: 1px solid red;
    /* no */
    border-radius: 20px;
    /* no */
}

.only-border::after {
    border-bottom: 1px solid red;
    /* no */
}

.px::after,.rem::after,.before::before,.after::after,.only-border::after {
                content: '';
                position: absolute;
                top: -50%;
                bottom: -50%;
                left: -50%;
                right: -50%;
                -webkit-transform: scale(0.5);
                        transform: scale(0.5);
                pointer-events: none;
            }"
`;
